pkg/scaffold/cmd.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaffold
+
+import (
+	"io"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+type cmd struct {
+	// ProjectPath is the project path rooted at GOPATH.
+	ProjectPath string
+}
+
+func renderCmd(w io.Writer, projectPath string) error {
```

Comment:
@fanminshi How about just making render a method on the struct rather than passing in all the values, and then initializing the struct to render it.
```Go
type Cmd struct {
	// CmdInput are all the input values for the Cmd template
	 CmdInput *CmdInput
}

type CmdInput struct {
	// ProjectPath is the project path rooted at GOPATH.
	ProjectPath string
}

func (cm * Cmd) renderTemplate(w io.Writer) error {
	t := template.New("main.go")

	t, err := t.Parse(mainTmpl)
	if err != nil {
		return err
	}

	return t.Execute(w, cm.CmdInput)
}

const main_tml = `package main
...
```

Reply:
that's fine with me.